packages/core/echo/echo/src/struggle.test.ts

```diff
@@ -0,0 +1,126 @@
+import { Schema } from 'effect';
+/// ISSUE(dm): Is it okay to have one entry point for all of the types like object id and type definitions and will it prohibit us from building isolated storybooks that don't pull in echo? DX vs tree shaking.
+// So that type and relation API should be separate from echo database API.
+// Potentially, some part of the database API could be included in there. Things like query builders. So only the declarative API, not the actual execution layer.
+import { test } from 'vitest';
+import { Type } from '.';
+
+declare const create: any;
+declare const Relation: any;
+declare const setupDb: any;
+declare const Filter: any;
+declare const Query: any;
+
+///
+
+const Contact = Schema.Struct({
+  // ISSUE(dm): Forcing explicit object IDs for every schema?
+  id: Type.ObjectId,
+  name: Schema.String,
+  // ISSUE(dm): Reconcile "type" and "typename" i.e. the dxn and the name of the type.
```

Comment:
i think `dxos.org/type/Contact` is the typename and `dxn:....:dxos.org/type/Contact` is the FQ typename. and Type is the object representation of the type.